freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -265,4 +281,8 @@ private String constructVaultPath(String accountId, String type, String subtype)
         return String.format("%s/%s/%s/%s-%s", accountId, type, subtype,
                 UUID.randomUUID().toString(), Long.toHexString(System.currentTimeMillis()));
     }
+
+    private static String constructCanonicalPrincipal(String serviceName, String hostName, String realm) {
```

Comment:
this duplicates the logic in `serviceAdd` method

Reply:
I wrongly resolved the merge conflict where rebasing the patch. Will address it.

Reply:
Addressed.